components/eamxx/src/doubly-periodic/impl/dp_iop_setinitial_impl.hpp

```diff
@@ -0,0 +1,207 @@
+#ifndef DP_IOP_SETINITIAL_IMPL_HPP
+#define DP_IOP_SETINITIAL_IMPL_HPP
+
+#include "dp_functions.hpp" // for ETI only but harmless for GPU
+
+#include "Context.hpp"
+#include "TimeLevel.hpp"
+
+namespace scream {
+namespace dp {
+
+/*
+ * Implementation of dp iop_setinitial. Clients should NOT
+ * #include this file, but include dp_functions.hpp instead.
+ */
+
+template<typename S, typename D>
+void Functions<S,D>::iop_setinitial(
+    const Int& plev,
+    const Int& pcnst,
+    const Int& nelemd,
+    const Int& np,
+    const Int& nstep,
+    const bool& use_replay,
+    const bool& dynproc,
+    const bool& have_t,
+    const bool& have_q,
+    const bool& have_ps,
+    const bool& have_u,
+    const bool& have_v,
+    const bool& have_numliq,
+    const bool& have_cldliq,
+    const bool& have_numice,
+    const bool& have_cldice,
+    const bool& scm_zero_non_iop_tracers,
+    const bool& is_first_restart_step,
+    const uview_1d<const Scalar>& qmin,
+    const uview_1d<const Spack>& uobs,
+    const uview_1d<const Spack>& vobs,
+    const uview_1d<const Spack>& numliqobs,
+    const uview_1d<const Spack>& numiceobs,
+    const uview_1d<const Spack>& cldliqobs,
+    const uview_1d<const Spack>& cldiceobs,
+    const Scalar& psobs,
+    const uview_1d<const Scalar>& dx_short,
+    Scalar& dyn_dx_size,
+    tracer_t& tracers,
+    element_t& elem,
+    const uview_1d<Spack>& tobs,
+    const uview_1d<Spack>& qobs)
+{
+  // Made these up
+
+  //FieldGroup g = ...;
+  //const auto& names = g.m_info->m_field_names;
+  constexpr Int inumliq = 1;
+  constexpr Int inumice = 2;
+  constexpr Int icldliq = 3;
+  constexpr Int icldice = 4;
+
+  const Int plev_packs = ekat::npack<Spack>(plev);
+  const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(1, plev_packs);
+
+  assert(np <= NP);
+  //assert(plev_packs <= Homme::NUM_LEV);
+  assert(tracers.inited());
+  assert(elem.inited());
+
+  // Get time info
+  const Int n0 = 0; //Homme::Context::singleton().get<Homme::TimeLevel>().n0;
```

Comment:
This attempt to use TimeLevel failed at runtime. I assume due to not being initialized:
`Error! Context member 'N5Homme9TimeLevelE' not found.`

Reply:
yes, homme has a routine to init the time level structure. For the `theta-l_kokkos` target, that's `init_time_level_c`.